lib/active_model/serializer/reflection.rb

```diff
@@ -106,17 +106,19 @@ def value(serializer)
       #
       # @api private
       #
-      def build_association(subject, parent_serializer_options)
-        association_value = value(subject)
+      def build_association(parent_serializer, parent_serializer_options)
+        association_value = value(parent_serializer)
         reflection_options = options.dup
-        serializer_class = subject.class.serializer_for(association_value, reflection_options)
+        serializer_class = parent_serializer.class.serializer_for(association_value, reflection_options)
```

Comment:
for just api-design, should a parent_serializer instance have a serialize_for method that just delegates to the class? would that semantically be weird?